test/common/test_point_type_static_member_functions.cpp

```diff
@@ -0,0 +1,74 @@
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   * Neither the name of the copyright holder(s) nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *
+ * $Id:  $
+ *
+ */
+
+#include <pcl/test/gtest.h>
+#include <pcl/pcl_tests.h>
+#include <pcl/point_types.h>
+
+using namespace pcl;
+using namespace pcl::test;
+
+TEST (PointTypeStaticMemberFunctions, DescriptorSizeTests)
+{
+  static_assert(PFHSignature125::descriptorSize() == 125, "incorrect descriptorSize");
+  static_assert(PFHRGBSignature250::descriptorSize() == 250, "incorrect descriptorSize");
+  static_assert(ShapeContext1980::descriptorSize() == 1980, "incorrect descriptorSize");
+  static_assert(UniqueShapeContext1960::descriptorSize() == 1960, "incorrect descriptorSize"); 
+  static_assert(SHOT352::descriptorSize() == 352, "incorrect descriptorSize");
+  static_assert(SHOT1344::descriptorSize() == 1344, "incorrect descriptorSize");
+  static_assert(FPFHSignature33::descriptorSize() == 33, "incorrect descriptorSize");
+  static_assert(VFHSignature308::descriptorSize() == 308, "incorrect descriptorSize");
+  static_assert(GRSDSignature21::descriptorSize() == 21, "incorrect descriptorSize");
+  static_assert(BRISKSignature512::descriptorSize() == 64, "incorrect descriptorSize");
+  static_assert(ESFSignature640::descriptorSize() == 640, "incorrect descriptorSize");
+  static_assert(GASDSignature512::descriptorSize() == 512, "incorrect descriptorSize");
+  static_assert(GASDSignature984::descriptorSize() == 984, "incorrect descriptorSize");
+  static_assert(GASDSignature7992::descriptorSize() == 7992, "incorrect descriptorSize");
+  static_assert(GFPFHSignature16::descriptorSize() == 16, "incorrect descriptorSize");
+  static_assert(Narf36::descriptorSize() == 36, "incorrect descriptorSize");
```

Comment:
These are just a TYPED_TEST masquerading as copy-paste.

https://github.com/google/googletest/blob/master/docs/advanced.md#typed-tests

Reply:
not sure i got it right (traits class especially), but here's a short list:

- could not use PCL_FEATURE_POINT_TYPES for sequence enumeration with BOOST_PP_SEQ_ENUM, so i made another list PCL_DESCRIPTOR_FEATURE_POINT_TYPES that consists from types that have descriptorSize() member function
- made traits template with specializations for all values of descriptorSize(), so that copy-pastes are eliminated